src/mempool/storage.rs

```diff
@@ -164,8 +164,21 @@ impl InMemoryStorage {
             self.proposal_will_wait(validator, car_proposal.clone());
             return ProposalVerdict::Wait(car_proposal.parent);
         }
+        debug!("Validating Car proposal: {:?}", car_proposal);
         for tx in car_proposal.txs.iter() {
             match &tx.transaction_data {
+                TransactionData::RegisterContract(tx) => {
+                    match node_state.handle_register_contract_tx(tx) {
+                        Ok(_) => {}
+                        Err(e) => {
+                            warn!(
+                                "Refusing Car Proposal: invalid RegisterContract transaction: {}",
+                                e
+                            );
+                            return ProposalVerdict::Refuse;
+                        }
+                    }
+                }
```

Comment:
Do you need that for the fix ?
cause it's kinda duplicating this PR : https://github.com/Hyle-org/hyle/pull/286

Reply:
j'avais effectivement des erreurs dans les logs des tests à cause de ça. Le CarProposal est refusé par node-2 car il peut pas vérifier la proof. Le test passe quand même car node-1 à lui-seul a le quorum nécessaire et peut envoyer un bloc avec les txs dedans même si node-2 les a refusés... il va falloir améliorer ça dans les tests, clairement

Donc en soit, je peux retirer cette partie là, les tests seront vert, mais c'est pas vraiment bon